keras/src/quantizers/gptq.py

```diff
@@ -0,0 +1,336 @@
+            elif d_model_dim_index in [1, 2]:  # Attention Output case
+                heads, head_dim, out_features = shape
+                self.rows, self.columns = heads * head_dim, out_features
+
+            # Create a temporary object that holds a reshaped
+            # 2D version of the kernel.
+            self.layer = type(
+                "temp",
+                (object,),
+                {
+                    "kernel": ops.reshape(
+                        layer.kernel, (self.rows, self.columns)
+                    ),
+                    "bias": layer.bias,
+                },
+            )()
+
+        else:
+            # Raise an error if the layer is not supported.
+            raise TypeError(f"Unsupported layer type for GPTQ: {type(layer)}")
+        self.H = ops.zeros((self.rows, self.rows), dtype="float32")
+
+    def update_hessian_with_batch(self, input_batch):
+        """
+        Updates the running average of the Hessian matrix with a new batch.
+
+        This method computes the Hessian matrix for a given batch of input
+        activations and updates the accumulated Hessian (`self.H`) using a
+        numerically stable running average. This allows the Hessian to be
+        computed over a large dataset without loading all samples into memory
+        at once.
+
+        The input tensor is first reshaped into a 2D matrix [num_samples,
+        num_features] before the Hessian is calculated.
+
+        Args:
+            input_batch: A 2D or higher-dimensional tensor of input activations
+                from a calibration batch.
+
+        Raises:
+            ValueError: If the feature dimension of the input tensor
+                `input_batch` does not match the dimensions of the
+                pre-initialized Hessian matrix `self.H`.
+        """
+        if input_batch is None:
+            raise ValueError("Input tensor 'input_batch' cannot be None.")
+
+        if len(input_batch.shape) < 2:
+            raise ValueError(
+                f"Input tensor 'input_batch' must have a rank of at least 2 "
+                f"(e.g., [batch, features]), but got rank "
+                f"{len(input_batch.shape)}."
+            )
+        if ops.size(input_batch) == 0:
+            raise ValueError("Input tensor 'input_batch' cannot be empty.")
+
+        if len(input_batch.shape) > 2:
+            input_batch = ops.reshape(input_batch, (-1, input_batch.shape[-1]))
+        input_batch = ops.cast(input_batch, "float32")
+
+        if self.H.shape[0] != input_batch.shape[-1]:
+            raise ValueError(
+                f"Hessian dimensions ({self.H.shape[0]}) do not"
+                "match input features ({input_batch.shape[-1]})."
+            )
+
+        current_hessian = ops.multiply(
+            2, ops.matmul(ops.transpose(input_batch), input_batch)
+        )
+
+        if self.num_samples == 0:
+            self.H = current_hessian
```

Comment:
rename `self.H` to `self.current_hessian` or at least `self.hessian`.

Reply:
Done